src/main/java/it/eng/idsa/businesslogic/util/AutoSelfRegistration.java

```diff
@@ -0,0 +1,60 @@
+package it.eng.idsa.businesslogic.util;
+
+import java.net.URISyntaxException;
+
+import javax.annotation.PreDestroy;
+import javax.xml.datatype.DatatypeConfigurationException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import de.fraunhofer.iais.eis.util.ConstraintViolationException;
+import it.eng.idsa.businesslogic.service.SelfDescriptionService;
+import it.eng.idsa.businesslogic.service.SelfRegistrationService;
+
+
+@ConditionalOnProperty(name="application.selfdescription.registrateOnStartup", havingValue="true")
+@Component
+public class AutoSelfRegistration{
+	
+	private static final Logger logger = LogManager.getLogger(AutoSelfRegistration.class);
+	
+	@Value("${application.selfdescription.brokerURL}")
+	private String brokerURL;
+	
+	@Autowired
+	private SelfDescriptionService selfDescriptionService;
+	
+	@Autowired
+	private SelfRegistrationService selfRegistrationService;
+	
+	@EventListener(ApplicationReadyEvent.class)
+	public void selfRegistrate() throws ConstraintViolationException, URISyntaxException, DatatypeConfigurationException {
+		logger.info("Starting AutoSelfRegistration");
+
+		selfRegistrationService.sendRegistrationRequest(selfDescriptionService.getConnectorAvailbilityMessage(),
```

Comment:
What would happen if we make this method without parameters, and move those parameters in selfRegistrationService.sendRegistrationRequest?
Those params are used only in that method - can we move them there?

Reply:
I left it like this in case we want to use this in the future for other requests, like how it's used here for sign up and sign out request. If you want i can make separate methods that send the corresponding message to one mutual method e.g. 
selfRegistrationService.registrate()(query,passivate..) {
sendRegistrationRequest(selfDescriptionService.getConnectorAvailbilityMessage(),
				selfDescriptionService.getConnectorAsString(), brokerURL);
}

Reply:
Other requests, if I'm not mistaking, are going through regular processors. This service is create to address this specific case - register on startup and unregister on shutdown. Maybe I'm not seeing bigger picture here :)
We can discuss this on call, if needed.

Reply:
Rename selfRegistrationService to BrokerService
rename sendRegistrationRequest method in selfRegistrationService to sendBrokerRequest
check if you can move brokerURL in BrokerService